design/use_cases/indoor_navigation_use_case.md

```diff
@@ -0,0 +1,14 @@
+# Indoor Navigation
+As a Robot user I want my robot to autonomously navigate to a given location on a given map so that it can help me at that location
+
+## More details
+- Why is this needed?
+   - This is needed for indoor robot navigation in most (all?) cases
+   - Example: a courier robot in a logistics warehouse
+
+- What is the expected user interaction?
+   - The user should be able to specify a map to use and a location on that map for the robot to move to. 
```

Comment:
Both: in global maps, it can be areas beyond the edge of the map, or areas within the center of the map for which we didn't have any observations during map building. In the current system, this is the "dark grey" portion of the map (free space is light grey, obstacles are black).

Reply:
What's the best way to handle unknown space? Is it to have the robot treat it as free space that can be traversed, as a keep-out area or something else? I think this one will require more thought / discussion. I do agree with your point however, that it needs to be handled in a consistent way. Maybe it needs to be a dynamically configurable way.

Reply:
It's highly dependent on the application -- hence why it is parameterized in the current system. I'm not suggesting we move away from allowing different modes of dealing with unknown space, I'm suggesting that the ways we deal with it are consistent throughout (ideally single parameter to select the mode of dealing with unknown space)

Reply:
As a pair of examples: in a warehouse which you have fully mapped, you probably don't want to allow navigation in unknown space. In an application where the robot is using navigation to explore the space and build a map, you almost certainly need to plan towards, and probably into, unknown space, to avoid start/stopping/replanning a lot.

Reply:
Somewhat related, the area enclosed by the edges of an obstacle (obstacle body so to speak) could be automatically painted as obstacle rather than unknown space. This would prevent e.g. simple discrete horizon-based planners from doing stupid things.